qiskit_experiments/library/randomized_benchmarking/layer_fidelity_unitary.py

```diff
@@ -0,0 +1,606 @@
+        :math:`EPLG = 1 - LF^{1/N_{2Q}}` where :math:`N_{2Q}` is number of 2-qubit gates in the layers.
+
+        Each of the 2-qubit (or 1-qubit) direct RBs yields the decaying probabilities
+        to get back to the ground state for an increasing sequence length (i.e. number of layers),
+        fits the exponential curve to estimate the decay rate, and calculates
+        the process fidelity of the subsystem from the rate.
+        LF is calculated as the product of the 2-qubit (or 1-qubit) process fidelities.
+        See Ref. [1] for details.
+
+        This unitary version allows artibrary 2Q gates
+
+    # section: analysis_ref
+        :class:`LayerFidelityAnalysis`
+
+    # section: reference
+        .. ref_arxiv:: 1 2311.05933
+
+    # section: example
+        .. jupyter-execute::
+            :hide-code:
+
+            # backend
+            from qiskit_aer import AerSimulator
+            from qiskit_aer.noise import NoiseModel, depolarizing_error
+
+            noise_model = NoiseModel()
+            noise_model.add_all_qubit_quantum_error(depolarizing_error(5e-3, 1), ["sx", "x"])
+            noise_model.add_all_qubit_quantum_error(depolarizing_error(0, 1), ["rz"])
+            noise_model.add_all_qubit_quantum_error(depolarizing_error(5e-2, 2), ["cx"])
+            backend = AerSimulator(noise_model=noise_model)
+
+        .. jupyter-execute::
+
+            import numpy as np
+            from qiskit import QuantumCircuit
+            from qiskit_experiments.library import StandardRB
+            from qiskit_experiments.library.randomized_benchmarking import LayerFidelityUnitary
+
+            lengths = np.arange(1, 80, 10)
+            two_qubit_layers=[[(0, 1), (3, 5)], [(1, 3), (5, 6)]]
+
+            num_samples = 3
+            seed = 106
+
+            qc = QuantumCircuit(2)
+            qc.rzz(0.5,0,1)
+
+            exp = LayerFidelityUnitary(
+                    physical_qubits=[0, 1, 3, 5, 6],
+                    two_qubit_layers=two_qubit_layers,
+                    lengths=lengths,
+                    backend=backend,
+                    num_samples=num_samples,
+                    seed=seed,
+                    two_qubit_gates=[qc.to_instruction()],
+            )
+
+            exp_data = exp.run().block_for_results()
+            results = exp_data.analysis_results(dataframe=True)
+
+            display(exp_data.figure(0)) # one of 6 figures
+            display(exp_data.analysis_results("EPLG", dataframe=True))
+
+            print(f"Available results: {set(results.name)}")
+    """
+
+    def __init__(
+        self,
+        physical_qubits: Sequence[int],
+        two_qubit_layers: Sequence[Sequence[Tuple[int, int]]],
+        lengths: Iterable[int],
+        backend: Optional[Backend] = None,
+        num_samples: int = 6,
+        seed: Optional[Union[int, SeedSequence, BitGenerator, Generator]] = None,
+        two_qubit_gates: Sequence[CircuitInstruction] = None,
```

Comment:
`to_instruction` actually returns an `Instruction` rather than a `CircuitInstruction`. `Instruction` is an abstract instruction that can be applied and `CircuitInstruction` is a specific application (it includes the qubits to apply the instruction on). Could we just make the type `Gate` though? That is a subclass of `Instruction`.

Reply:
Ah ok, that probably makes more sense. And then it would be from a circuit using `.to_gate()`?

Reply:
I don't think it would change other code. If we only supported a named gate and the output of `QuantumCircuit.to_instruction`/`QuantumCircuit.to_gate`, I think we could take out the change to `clifford_utils.py` where a special case for instructions with names starting with "circuit" was added. I think that is falling into the case of adding internal instructions one by one instead of just `data.append` like it could if `RZZGate` was passed directly.